shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -338,6 +338,24 @@ public void setupConfiguration(Configuration conf) {
       conf.setInt(MRJobConfig.REDUCE_MEMORY_MB, 512);
       conf.setInt(MRJobConfig.MR_AM_VMEM_MB, 128);
     }
+
+    // we want this shim to override these values only if the default value is set, otherwise there
+    // is a good chance that the user of the minicluster set another value intentionally
+    protected void overrideIntIfDefaultIsSet(Configuration conf, String key, int defaultVal,
+        int newVal) {
+      if (conf.getInt(key, defaultVal) == defaultVal) {
+        LOG.info("Hadoop23Shims overrides '{}' from {} to {}", key, defaultVal, newVal);
+        conf.setInt(key, newVal);
+      }
```

Comment:
Would it make sense to LOG even if we do not override?
It took me several attempts to understand your comments in the description (I have kept looking for 256 in the logs), so either it is too early morning for me or some additional log messages would be good 😄 

If I understand correctly this is test code only change so it would not be a problem if we add more log.

Your thoughts? And BTW thanks for the patch!
Peter

Reply:
right, I'm adding logs for untouched configs too

Reply:
added in second commit: https://github.com/apache/hive/pull/1941/commits/697dc4a9e37c12b4d7d34589cc1014d6e08a6700

edited PR description with new log contents and some instructions